### DISCUSSION TITLE
Schedule timer support of seconds

### DISCUSSION THREAD

**Discussion by KG3RK3N** [emsesp/EMS-ESP32#1566] [General]
Hello, I have created a timer to set a gpio pin with a value of the boiler.
But currently, I can set the timer to only 1 minute or higher. But I want to set it to like 10 seconds.

Is it possible in the future to allow seconds on the scheduler?

![image](https://github.com/emsesp/EMS-ESP32/assets/16717995/0c6f5032-50ed-4d49-9121-3e71cf228968)

**Comment by MichaelDvP** [comment]
It makes no sense, most telegrams on the ems-bus are only send once a minute.
Heating is a slow process and you will never feel a difference if heating starts a minute later.

**Comment by proddy** [comment]
I agree, although see @KG3RK3N's use case here. The heating active comes from the UBAMonitorFast telegram which is broadcasted every 10 seconds.

**Comment by KG3RK3N** [comment]
It's not so important, but if it's easy to implement I think why not support that and restrict the user?
Sure, if the implementation of that is complex then it makes no sense to spend much time.

**Reply by MichaelDvP** [reply]
Changing minute checking to second checking is easy to implement. 
But it is very, very complex to avoid users sending ems commands in short timings and flooding the ems-tx-queue. This will block ems-esp from fetching ems-values.
The bus is slow and we can send only telegrams if the master polls, that can be seconds from one to the next. 
We already had this, that a user sends mqtt commands very fast and blocks ems-esp from updating the values.

**Reply by proddy** [reply]
that's very true.

Your use case @KG3RK3N, I think, is to get closer to real-time info on whether the heating is on or off. Maybe we can look into another way of solving that for you. Today, there's an MQTT topic with that data, which is updated frequently within a 10-second window. Can that work for you?

**Reply by MichaelDvP** [reply]
@KG3RK3N If you like to try to see the difference of minute/second and can compile your own build: Just remove the `/60` from https://github.com/emsesp/EMS-ESP32/blob/1b27e1fd09474a0f09a3f8088b8b8e9d89a4d3c2/src/web/WebSchedulerService.cpp#L394 and the Timer value is interpreted as `mm:ss` (Time-schedule is still hh:mm).

**Reply by KG3RK3N** [reply]
Ah perfect, that's easy. I will try that.

I understand the problem with the flooding.
The solution with mqtt is a good idea, but I want to control the pin without wifi. So if the wifi connection is lost, and I had that already some times, the value can't update and the pumps are still on or off. 

Any plans to have triggers like that without time scheduling? So that after a value has changed of a given property, a action can triggered like set gpio pin?
That should prevent the flooding of some users, I guess.

**Reply by MichaelDvP** [reply]
> Any plans to have triggers like that without time scheduling?

Trigger on a change of an entity should be possible, we already have this for mqtt publish on change. But i think this will go strait to #1480. 
Next is that we should trigger on a compare like `temperaturesensor/outdoor < 5` or with compare entities with math `temperaturesensor/outdoor < temperaturesensor/other + 3`.
and at last we need settings of more complex calculations like building outdoorcontroled heating e.g.:
cmd: `boiler/selflowtemp`, value: `(20 - temperatursensor/outdoor/value) * 2.5 + 1`
This needs a good parser, something like the USE_EXPRESSION section of [tasmota](https://github.com/arendst/Tasmota/blob/development/tasmota/tasmota_xdrv_driver/xdrv_10_rules.ino). A lot of work. Don't expect it in the near future.

**Reply by proddy** [reply]
The conditional logic using expressions would be really nice, but hard and complicated to code for now. Still an interesting idea so please @KG3RK3N add your thoughts to #1480 like Michael mentioned.

The entity change logic exists in the code, so we could extend to the scheduler. We would need to build in some failsafes so it's throttled, and there are no deadlocks from infinite loops, etc. But fun to try out.
